pkg/git/git_test.go

```diff
@@ -0,0 +1,221 @@
+package git
+
+import (
+	"fmt"
+	"net/http/httptest"
+	"os"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/sosedoff/gitkit"
+	"github.com/stretchr/testify/require"
+
+	libOS "github.com/akuity/kargo-render/internal/os"
+)
+
+func TestRepo(t *testing.T) {
+	testRepoCreds := RepoCredentials{
+		Username: "fake-username",
+		Password: "fake-password",
+	}
+
+	// This will be something to opt into because on some OSes, this will lead
+	// to keychain-related prompts.
+	useAuth, err := libOS.GetBoolFromEnvVar("TEST_GIT_CLIENT_WITH_AUTH", false)
+	require.NoError(t, err)
+	service := gitkit.New(
+		gitkit.Config{
+			Dir:        t.TempDir(),
+			AutoCreate: true,
+			Auth:       useAuth,
+		},
+	)
+	require.NoError(t, service.Setup())
+	service.AuthFunc =
+		func(cred gitkit.Credential, _ *gitkit.Request) (bool, error) {
+			return cred.Username == testRepoCreds.Username &&
+				cred.Password == testRepoCreds.Password, nil
+		}
+	server := httptest.NewServer(service)
+	defer server.Close()
+
+	testRepoURL := fmt.Sprintf("%s/test.git", server.URL)
+
+	rep, err := Clone(testRepoURL, testRepoCreds)
+	require.NoError(t, err)
+	require.NotNil(t, rep)
+	r, ok := rep.(*repo)
+	require.True(t, ok)
+
+	t.Run("can clone", func(t *testing.T) {
```

Comment:
Nit: these test cases indirectly depend on each other, which makes running a specific test case in isolation hard. Avoiding this in the future would be great.

Reply:
Actually, they don't... anything that is an absolute requirement for each test case to succeed is done between test cases.

Reply:
TIL this works flawlessly.